mlrun/serving/v2_serving.py

```diff
@@ -586,28 +587,35 @@ def _init_endpoint_record(
             model_endpoint=model_endpoint.dict(),
         )
 
-    elif (
-        model_ep
-        and (
+    elif model_ep:
+        attributes = {}
+        old_model_uri = model_ep.spec.model_uri
+        mlrun.model_monitoring.helpers.enrich_model_endpoint_with_model_uri(
+            model_endpoint=model_ep,
+            model_obj=model.model_spec,
+        )
+        if model_ep.spec.model_uri != old_model_uri:
+            attributes["model_uri"] = model_ep.spec.model_uri
+        if (
             model_ep.spec.monitoring_mode
             == mlrun.common.schemas.model_monitoring.ModelMonitoringMode.enabled
-        )
-        != model.context.server.track_models
-    ):
-        monitoring_mode = (
-            mlrun.common.schemas.model_monitoring.ModelMonitoringMode.enabled
-            if model.context.server.track_models
-            else mlrun.common.schemas.model_monitoring.ModelMonitoringMode.disabled
-        )
-        db = mlrun.get_run_db()
-        db.patch_model_endpoint(
-            project=project,
-            endpoint_id=uid,
-            attributes={"monitoring_mode": monitoring_mode},
-        )
-        logger.debug(
-            f"Updating model endpoint monitoring_mode to {monitoring_mode}",
-            endpoint_id=uid,
-        )
+        ) != model.context.server.track_models:
+            attributes["monitoring_mode"] = (
+                mlrun.common.schemas.model_monitoring.ModelMonitoringMode.enabled
+                if model.context.server.track_models
+                else mlrun.common.schemas.model_monitoring.ModelMonitoringMode.disabled
+            )
+        if attributes:
+            db = mlrun.get_run_db()
+            db.patch_model_endpoint(
+                project=project,
+                endpoint_id=uid,
+                attributes={"model_uri": model_ep.spec.model_uri},
```

Comment:
Shouldn't it be:
```suggestion
                attributes=attributes,
```
?